### PR TITLE
Make Notes on Volunteers Delete-able

### LINKED ISSUE #3405: Make Notes on Volunteers Delete-able
**What type(s) of user does this feature affect?**
 - supervisors
 - admins

**Description**
Currently, Supervisors and Admins can make and edit notes on Volunteers, but those notes cannot be deleted. We would like to add a delete button and the ability to delete Notes. Notes on Volunteers can be found on the Editing Volunteer page when logged in as a Supervisor or Admin.

**Screenshots of current behavior, if any**  
Current Volunteer Note State

_QA Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 6-8pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by iraline**
### What github issue is this PR for, if any?
Resolves #3405

### What changed, and why?
To be possible to admin or supervisor delete notes from volunteers, a button and a method are needed to do it.

### How will this affect user permissions?
- Volunteer permissions: N/A
- Supervisor permissions: N/A
- Admin permissions: N/A

### How is this tested? 
Request spec - Delete route is tested
System spec - Checks if the notes appear in the correct amount and, when deleting one, it disappears from the page

### Feelings gif 
casa

### CODE REVIEW

File: app/views/volunteers/_notes.html.erb [hunk 1]
```diff
@@ -22,6 +22,7 @@
                 <td><%= note.created_at.strftime(t("time.formats.standard")) %></td>
                 <td>
                   <%= link_to "Edit", edit_volunteer_note_path(@volunteer, note), class: "btn btn-primary" %>
+                  <%= link_to "Delete", volunteer_note_path(@volunteer, note), class: "btn btn-primary", method: :delete %>
```
Comment (m-pereira) [inline app/views/volunteers/_notes.html.erb:25]: wondering if would be good to have confirmation here 🤔

**Comment by iraline** [comment]
I will check these errors that appeared right after the merge

**Comment by iraline** [comment]
I ran the tests locally and they all passed 🤔 .
Should I do something about it (like closing this PR and open again to see if works) or will be automatic correctly?

image

**Comment by m-pereira** [comment]
> I ran the tests locally and they all passed thinking . Should I do something about it (like closing this PR and open again to see if it works) or will be fixed automatically? image

I've run the failing tests again, let's see

File: app/views/volunteers/_notes.html.erb (hunk 1 above)
Comment (compwron) [reply app/views/volunteers/_notes.html.erb:25]: good point... or maybe just "soft" delete i.e. mark it deleted but don't actually delete it all the way...?

**Comment by compwron** [comment]
hmm I will take a look...

**Comment by compwron** [comment]
Ci failures:
```
rspec ./spec/system/volunteers/edit_spec.rb:425 # volunteers/edit logged in as a supervisor can delete notes about a volunteer
rspec ./spec/system/volunteers/edit_spec.rb:381 # volunteers/edit logged in as an admin can delete notes about a volunteer
```

**Comment by iraline** [comment]
@compwron Can I help with anything else on this issue?

**Review by compwron — APPROVED** [review]

**Comment by compwron** [comment]
Yay!!

**Pull request merged**